components/post_view/channel_intro_message/boards_button/boards_button.test.tsx

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+
+import BoardsButton from './boards_button';
+
+const component = (otherProps: any) => {
```

Comment:
Any particular reason for using this pattern for testing a component over the one where we have `baseProps` defined and shared that between the test cases?

Reply:
When I was first working on the conversion, I had initially kept all tests in the `ChannelIntroComponent` test and was checking one child deep because of the refactor.  I converted some to use baseProps later and didn't convert some of tests.  I like the baseProps usage and will convert all test to use that method.  Thanks for catching!

Reply:
After looking at it a bit more, having the component defined like this allows reducing the number of lines per test because you don't need the <BoardsButton> instance for each test.  I'd prefer this pattern because each test file is only testing one component.

Reply:
Sounds good. I'd usually tend towards consistency with other code, but in this case, there's a good reason not to